src/main/java/Duke.java

```diff
@@ -1,10 +1,276 @@
+import java.util.Scanner;
+import java.util.List;
+import java.util.ArrayList;
+
+class Task {
+    private String message;
+    private boolean isDone;
+
+    Task(String message) {
+        this.message = message;
+        this.isDone = false;
+    }
+
+    public String getMessage() {
+        return message;
+    }
+
+    public String getStatusIcon() {
+        return "[" + (isDone ? "\u2713" : "\u2718") + "] ";
+    }
+
+    public void setDone() {
+        this.isDone = true;
+    }
+
+    public String getTypeLetter() {
+        // dummy value
+        return "";
+    }
+}
+
+class UnknownCommandException extends IndexOutOfBoundsException {
```

Comment:
Should these exception classes be moved into separate files for better abstraction?